tests/lib/ocpp/v201/test_smart_charging_handler.cpp

```diff
@@ -686,7 +686,11 @@ TEST_F(ChargepointTestFixtureV201, K01FR44_IfPhaseToUseProvidedForDCChargingStat
     EXPECT_THAT(sut, testing::Eq(ProfileValidationResultEnum::ChargingSchedulePeriodExtraneousPhaseValues));
 }
 
-TEST_F(ChargepointTestFixtureV201, K01FR45_IfNumberPhasesGreaterThanMaxNumberPhasesForACEVSE_ThenProfileIsInvalid) {
+TEST_F(ChargepointTestFixtureV201,
+       K01FR45_IfNumberPhasesGreaterThanChargingStationSupplyPhasesForACEVSE_ThenProfileIsInvalid) {
+    device_model->set_value(ControllerComponentVariables::ChargingStationSupplyPhases.component,
+                            ControllerComponentVariables::ChargingStationSupplyPhases.variable.value(),
+                            AttributeEnum::Actual, std::to_string(0), "test", true);
```

Comment:
When going over the requirements for FR.45 again, we realized that we misinterpreted the check that it was asking for. This change is a correction - we compare the number of phases in the profile and reject it if it's greater than what the charging station can supply (AKA ChargingStatisonSupplyPhases). 

Technically 0 is not realistic, but we want to have invalid values in any case. As long as the amount of `SupplyPhases` is less than the given `numberPhases`, we're testing this use case.
 
We don't think a comment is necessary for correcting behavior that was incorrect to begin with.

Reply:
Just to clarify for the future: this would be a good comment to put into the commit message that changed the check and to flag in the PR description. Again, the git history (`annotate` in the CLI and `blame` in the UI) will flag that this was changed in a different commit than the rest of the tests. Future community members may want to see why.

Reply:
I will try to ensure the individual commit messages are a bit more documenting going forward. It is valuable information to include.